client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.getElementsByClassName( 'banner__description"' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should render a .banner__list when list is specified', () => {
-		const comp = shallow( <Banner { ...props } list={ [ 'test1', 'test2' ] } /> );
-		expect( comp.find( '.banner__list' ) ).toHaveLength( 1 );
-		expect( comp.find( '.banner__list li' ) ).toHaveLength( 2 );
-		expect( comp.find( '.banner__list li' ).at( 0 ).text() ).toContain( 'test1' );
-		expect( comp.find( '.banner__list li' ).at( 1 ).text() ).toContain( 'test2' );
+		const { container } = render( <Banner { ...props } list={ [ 'test1', 'test2' ] } /> );
+		expect( container.querySelectorAll( '.banner__list' ) ).toHaveLength( 1 );
+		expect( container.querySelectorAll( '.banner__list li' ) ).toHaveLength( 2 );
+		expect( screen.getByText( 'test1' ) ).toBeVisible();
+		expect( screen.getByText( 'test2' ) ).toBeVisible();
 	} );
 
 	test( 'should not render a .banner__list when description is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner__list' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.getElementsByClassName( '.banner__list' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should record Tracks event when event is specified', () => {
-		const comp = shallow( <Banner { ...props } event="test" /> );
-		expect( comp.find( 'TrackComponentView' ) ).toHaveLength( 1 );
+		renderWithProvider( <Banner { ...props } event="test" /> );
+		expect( recordTracksEvent.mock.calls[ 0 ][ 1 ].cta_name ).toBe( 'test' );
 	} );
 
 	test( 'should not record Tracks event when event is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( 'TrackComponentView' ) ).toHaveLength( 0 );
+		renderWithProvider( <Banner { ...props } /> );
+		expect( recordTracksEvent ).not.toHaveBeenCalled();
 	} );
 
 	test( 'should render Card with href if href prop is passed', () => {
-		const comp = shallow( <Banner { ...props } href={ '/' } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBe( '/' );
+		const { container } = render( <Banner { ...props } href={ '/' } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).toHaveAttribute( 'href', '/' );
 	} );
 
 	test( 'should render Card with no href if href prop is passed but disableHref is true', () => {
-		const comp = shallow( <Banner { ...props } href={ '/' } disableHref={ true } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( Card ).props().href ).toBeNull();
+		const { container } = render( <Banner { ...props } href={ '/' } disableHref={ true } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).not.toHaveAttribute( 'href', '/' );
```

Comment:
Should we remove the `'/'` from here, which would more generally ensure we don't have a `href` at all?

Reply:
Remove the prop? From: `render( <Banner { ...props } href="/" disableHref={ true } /> );`

Or from the assertion? `expect( container.firstChild ).not.toHaveAttribute( 'href', '/' );`

Reply:
I mean removing it from the assertion. Because the way we've written the assertion, it will pass if:

* no `href` is rendered
* `href` is different than `/`

and it seems to me that we want only the first one to be true, and if the second is the case, the test should fail.